src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -37,13 +37,15 @@ public partial class HostBuilder : IHostBuilder
         private HostingEnvironment? _hostingEnvironment;
         private IServiceProvider? _appServices;
         private PhysicalFileProvider? _defaultProvider;
+        private readonly bool? _defaultProviderFactoryUsed;
```

Comment:
Why a nullable here? I don't see where it matters whether this is `false` vs. `null`.